libraries/apollo-compiler/src/main/kotlin/com/apollographql/apollo3/compiler/ir/SelectionSetsBuilder.kt

```diff
@@ -79,19 +80,35 @@ internal class SelectionSetsBuilder(
     val fieldDefinition = definitionFromScope(schema, parentType)!!
 
     val selectionSetName = resolveNameClashes(usedNames, name)
+
+    /**
+     * Pull all arguments from the schema as we need them to compute the cache key
+     */
+    val typeDefinition = schema.typeDefinition(parentType)
+    val actualArguments = fieldDefinition.arguments.map { schemaArgument ->
+      val operationArgument = arguments.firstOrNull { it.name == schemaArgument.name }
+
+      val keyArgs = typeDefinition.keyArgs(name, schema)
+      val paginationArgs = typeDefinition.paginationArgs(name, schema)
+
+      /**
+       * When passed explicitly, the argument values are coerced (but not their default value)
+       */
+      val userValue = operationArgument?.value?.coerceInExecutableContextOrThrow(schemaArgument.type, schema)
+      IrArgument(
+          name = schemaArgument.name,
+          value = (userValue ?: schemaArgument.defaultValue)?.toIrValue(),
+          isKey =  keyArgs.contains(schemaArgument.name),
+          isPagination = paginationArgs.contains(schemaArgument.name)
+      )
+    }
```

Comment:
This is the other important change. The `CompiledSelections` now contain even more schema information.